app/router.js

```diff
@@ -2,7 +2,11 @@ import Ember from "ember";
 import config from "./config/environment";
 
 var Router = Ember.Router.extend({
-  location: config.locationType
+  location: config.locationType,
+  clearMessages: function() {
+    // Clear any sticky flash messages when leaving a route
+    Ember.get(this, 'flashMessages').clearMessages();
+  }.on('didTransition')
```

Comment:
`didTransition` also bubbles through routes. This might be better as a hook on `actions:` for the application route:

```
export default Ember.Route.extend({
  // some application route stuff
  actions: {
    didTransition() {
      // clear the messages
    }
```